tests/cases.py

```diff
@@ -1938,3 +1939,67 @@ def test_score_h(self):
         scores = self.instance.score_h(rt_batch=rt_batch)
         assert scores.shape == (self.batch_size, self.factory.num_entities)
         self._verify(scores)
+
+
+class RankBasedMetricTestCase(unittest_templates.GenericTestCase[RankBasedMetric]):
+    """A test for rank-based metrics."""
+
+    #: the maximum number of candidates
+    max_num_candidates: int = 17
+
+    #: the number of ranks
+    num_ranks: int = 33
+
+    #: the number of candidates for each individual ranking task
+    num_candidates: numpy.ndarray
+
+    #: the ranks for each individual ranking task
+    ranks: numpy.ndarray
+
+    def post_instantiation_hook(self) -> None:
+        """Generate a coherent rank & candidate pair."""
+        generator = numpy.random.default_rng()
+        self.num_candidates = generator.integers(low=1, high=self.max_num_candidates, size=(self.num_ranks,))
```

Comment:
can we put utilities for generating random rankings in some reusable utility functions

Reply:
2e96a36 - not yet moved into a separate module